### PR TITLE
samples: synchronization: updated code and documentation for SMP

### PR DESCRIPTION
**Pull request by youndong**
When samples/synchronization is executed on a board
that supports SMP (eg qemu_x86_64), it shows the execution
results of threads running on different cpu.

Signed-off-by: Youndong Park <youndong.park@gmail.com>

### CODE REVIEW

**Review by youndong — COMMENTED** [review]
Updated the code and documentation on synchronization sample. README.rst shows the actual results when the binary runs.

File: samples/synchronization/README.rst [hunk 1]
```diff
@@ -30,17 +30,32 @@ Sample Output
...
-   threadB: Hello World!
-   threadA: Hello World!
-   threadB: Hello World!
-   threadA: Hello World!
-   threadB: Hello World!
-   threadA: Hello World!
-   threadB: Hello World!
-   threadA: Hello World!
-   threadB: Hello World!
+   threed_a: Hello World from cpu(0) on qemu_x86!
```
Comment (stephanosio) [inline samples/synchronization/README.rst:33]: typo?

File: samples/synchronization/src/main.c [hunk 2]
```diff
@@ -127,6 +127,10 @@ void main(void)
 	k_thread_cpu_mask_enable(&threadB_data, 1);
 #endif
 
+#if CONFIG_SMP
```
Comment (stephanosio) [inline samples/synchronization/src/main.c:130]:
```suggestion
#ifdef CONFIG_SMP
```

**Review by stephanosio — CHANGES_REQUESTED** [review]
Please update the commit title to match that of the PR.

File: samples/synchronization/src/main.c (hunk 2 above)
Comment (youndong) [reply samples/synchronization/src/main.c]: Thank you for your correction.

**Review by youndong — COMMENTED** [review]
@stephanosio, I updated some code and README.rst according to your comments. Thank you.

**Pull request closed without merging**